back/app/controllers/application_controller.rb

```diff
@@ -82,6 +82,17 @@ def linked_json(collection, serializer, options = {})
     }
   end
 
+  def serialize_heterogeneous_collection(collection)
+    collection.map { |object| serialize_by_type(object) }
+  end
+
+  # This will only work if the serializer for the object is in the WebApi::V1 namespace,
+  # is named <RecordType>Serializer, and has a type field with value <RecordType> in its attributes.
+  def serialize_by_type(object)
+    serializer_class = "WebApi::V1::#{object.type}Serializer".constantize
+    serializer_class.new(object, params: fastjson_params).serializable_hash[:data]
+  end
```

Comment:
If we do go this route, this should probably be able to infer the correct namespace, and not rely on the serializers always being in the `WebApi::V1`  namespace (e.g. could be in an engine module).